frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/ApplicationMaster.java

```diff
@@ -753,16 +672,36 @@ private void recoverTransitionTaskStateQueue() {
   private void addContainerRequest(TaskStatus taskStatus) throws Exception {
     String taskRoleName = taskStatus.getTaskRoleName();
     TaskStatusLocator taskLocator = new TaskStatusLocator(taskRoleName, taskStatus.getTaskIndex());
-    ContainerRequest request = setupContainerRequest(taskStatus);
+    Integer containerRequestTimeoutSec = CommonUtils.getRandomNumber(
+        conf.getLauncherConfig().getAmContainerRequestMinTimeoutSec(),
+        conf.getLauncherConfig().getAmContainerRequestMaxTimeoutSec());
 
-    LOGGER.logInfo("%s: addContainerRequest: %s", taskLocator, HadoopExtensions.toString(request));
+    ContainerRequest request = setupContainerRequest(taskStatus);
+    LOGGER.logInfo("%s: addContainerRequest with timeout %ss. ContainerRequest: [%s]",
+        taskLocator, containerRequestTimeoutSec, HadoopExtensions.toString(request));
     rmClient.addContainerRequest(request);
-    statusManager.transitionTaskState(taskLocator, TaskState.CONTAINER_REQUESTED);
+    statusManager.transitionTaskState(taskLocator, TaskState.CONTAINER_REQUESTED,
+        new TaskEvent().setContainerRequest(request));
+
+    transitionTaskStateQueue.queueSystemTaskDelayed(() -> {
+      if (statusManager.containsTask(request.getPriority())) {
+        LOGGER.logInfo(
```

Comment:
>logInfo [](start = 15, length = 7)

logwarm?

Reply:
OK

---
In reply to: [158203701](https://github.com/Microsoft/pai/pull/121#discussion_r158203701) [](ancestors = 158203701)